compiler/luci-interpreter/src/kernels/FullyConnected.cpp

```diff
@@ -188,5 +205,33 @@ void FullyConnected::evalQuantizedS8() const
     getTensorShape(output()), getTensorData<int8_t>(output()));
 }
 
+void FullyConnected::evalHybrid() const
+{
+  float activation_min{};
+  float activation_max{};
+  calculateActivationRange(_params.activation, &activation_min, &activation_max);
+
+  tflite::FullyConnectedParams params{};
+  params.float_activation_min = activation_min;
+  params.float_activation_max = activation_max;
+  params.weights_format = tflite::FullyConnectedWeightsFormat::kDefault;
+
+  const int8_t *weights_packed = getTensorData<int8_t>(weights());
+  float *weights_float = getTensorData<float>(scratch());
+  const Shape &weights_shape = weights()->shape();
+  for (uint32_t i = 0; i < (weights_shape.num_elements() + 1) / 2; ++i)
+  {
+    uint32_t idx = i * 2;
+    // 1bit for sign, 3bit for value
+    weights_float[idx] = weights()->scale() * (static_cast<int8_t>(weights_packed[i] << 4) >> 4);
+    if (idx < weights_shape.num_elements())
+      weights_float[idx + 1] = weights()->scale() * (static_cast<int8_t>(weights_packed[i]) >> 4);
+  }
```

Comment:
```suggestion
  for (uint32_t i = 0; i < weights_shape.num_elements(); ++i)
  {
    weights_float[i] = weights()->scale() * (static_cast<int8_t>(weights_packed[i] << 4) >> 4);
  }
```

If we consider input weight as not-packed 4bits (which has 8bit data type), we do not need to consider unpacking.